### PR TITLE
Add Provider and Flavor enums, populate their default values in ProvisionConfigInstance

### PR DESCRIPTION
**Pull request by engechas**
### Description
Adds the new Provider and Flavor models for the build system expansion.
#132 

Signed-off-by: Chase Engelbrecht <engechas@amazon.com>
 
### Check List
- [x] New functionality includes testing
  - [x] All unit and integration tests pass
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/builder/models/providers.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from enum import Enum
+
+
+class Provider(str, Enum):
```
Comment (travisbenedict) [inline osbenchmark/builder/models/providers.py:4]: How do you feel about calling this something like `InfraProvider` or `ClusterInfraProvider` instead?

File: osbenchmark/builder/models/flavors.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from enum import Enum
+
+
+class Flavor(str, Enum):
```
Comment (travisbenedict) [inline osbenchmark/builder/models/flavors.py:4]: Thoughts on calling this `ClusterFlavor` instead?

File: osbenchmark/builder/models/providers.py (hunk 1 above)
Comment (engechas) [reply osbenchmark/builder/models/providers.py:4]: That's more clear, will change

File: osbenchmark/builder/models/flavors.py (hunk 2 above)
Comment (engechas) [reply osbenchmark/builder/models/flavors.py:4]: That's more clear, will change

**Review by travisbenedict — APPROVED** [review]

File: osbenchmark/builder/models/cluster_flavors.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from enum import Enum
+
+
+class ClusterFlavor(str, Enum):
+    SELF_MANAGED = "self_managed"
+    MANAGED = "managed"
```
Comment (achitojha) [inline osbenchmark/builder/models/cluster_flavors.py:6]: Do we need two different enums for ClusterFlavor & ClusterInfraProvider? Can you explain what's the use case supporting the need for two enums.

I was thinking of combining them into one. The `MANAGED` flavor seems too generic for me, similarly `LOCAL` provider probably doesn't add much value.

**Review by IanHoang — APPROVED** [review]
LGTM

**Pull request merged**